test/integration/admin/api/personal/notification_preferences_controller_test.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class Admin::Api::Personal::NotificationPreferencesControllerTest < ActionDispatch::IntegrationTest
+  def setup
+    provider = FactoryBot.create(:provider_account)
+    @user = provider.admin_users.first!
+    @token = FactoryBot.create(:access_token, owner: @user, scopes: %w[account_management]).value
+    host! provider.external_admin_domain
+  end
+
+  attr_reader :user, :token
+
+  test 'index' do
+    get admin_api_personal_notification_preferences_path(format: :json, access_token: @token)
+
+    prefs =  JSON.parse(response.body)['notification_preferences']
+
+    assert_response :success
+    assert_same_elements NotificationPreferences.available_notifications, prefs.keys.map(&:to_sym)
+    assert prefs.values.all? { [true, false].include?(_1) }
+  end
+
+  test 'update successfully' do
```

Comment:
I think we should add another test for the categories restriction I explained above.